### DISCUSSION TITLE
Log panel enhancement

### DISCUSSION THREAD

**Discussion by tktr** [grafana/grafana#50049] [Dashboards, Panels & Visualizations]
With the current state of the **log panel**, it is not possible to have a usable preview of a document if more than one field is required. What would be needed is exactly the same behavior as in the explore area with a possibility to toggle different fields:

![image](https://user-images.githubusercontent.com/12617424/171507295-f5102d54-5cb6-495b-8d50-0019b5982604.png)

I also tried to find a workaround with a combination of the `Add field from calculation` and `Organize fields` transformations but the calculated field is not picked up by the log panel.

This is similar to discussions already raised in other issues like #30817 and [#22271](https://github.com/grafana/grafana/issues/22271#issuecomment-590893570) where a general enhancement of the log panel user experience is requested. For us this is currently a blocker to fully onboard log based dashboards into our Grafana instance.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
